### DISCUSSION TITLE
How to get CollidablePair (CollidableReferences) from CollisionQuery result?

### DISCUSSION THREAD

**Discussion by asimplestd** [bepu/bepuphysics2#137] [Q&A] [answered]
Hello, I got a question related to CollisionQuery functionality!
CollisionQueryDemo uses CollisionBatcher for collision queries. CollisionBatcher uses shapes and poses directly with no reference to  CollidableReference. BatcherCallbacks (ICollisionCallbacks) do not have any CollidableReference related info.
Is it possible get CollidablePair info of 'collided queries' ?
Many thanks in advance!

**Comment by RossNordby** [comment, marked as answer]
When adding a job to the `CollisionBatcher`, you provide a `pairId`. When the `CollisionBatcher` finishes a test, it'll report the results alongside the `pairId` associated with the test. That can be used to look up other information. The `CollisionBatcher` is otherwise blind to concepts like collidables and collidable references.

Often, this will look like setting up a list of the stuff you want to look up while you're adding tests to the collision batcher. In the `CollisionQueryDemo`, this is the `queryWasTouched` buffer used to color the shapes, but it could contain any data.

**Reply by asimplestd** [reply]
Got it, thanks!
